media/js/base/datalayer-productdownload.es6.js

```diff
@@ -95,38 +109,101 @@ TrackProductDownload.getEventFromUrl = (downloadURL) => {
         let platform = params.os;
         platform = platform === 'osx' ? 'macos' : platform;
         // release channel is second word of product param
-        // (except for release which says 'latest' but we want 'release')
+        // (except for latest or sub installer - we update those to say release)
         const releaseChannel =
-            productSplit[1] === 'latest' ? 'release' : productSplit[1];
+            productSplit[1] === ('latest' || 'stub')
+                ? 'release'
+                : productSplit[1];
 
         eventObject = TrackProductDownload.getEventObject(
             product,
             platform,
+            'direct',
+            downloadURL,
             releaseChannel,
             params.lang
         );
     } else if (playStoreURL.test(downloadURL)) {
         const idParam = params.id;
-        let androidRelease = 'release';
-        // Android Play Store, need to check release, beta, nightly
+        let androidProduct = 'unrecognized';
+        let androidRelease = '';
+
         switch (idParam) {
+            case 'org.mozilla.firefox':
+                androidProduct = 'firefox';
+                androidRelease = 'release';
+                break;
             case 'org.mozilla.fenix':
+                androidProduct = 'firefox';
                 androidRelease = 'nightly';
                 break;
             case 'org.mozilla.firefox_beta':
+                androidProduct = 'firefox';
                 androidRelease = 'beta';
                 break;
+            case 'org.mozilla.focus':
+                androidProduct = 'focus';
+                break;
+            case 'org.mozilla.klar':
+                androidProduct = 'klar';
+                break;
+            case 'com.ideashower.readitlater.pro':
+                androidProduct = 'pocket';
+                break;
         }
+
         eventObject = TrackProductDownload.getEventObject(
-            'firefox',
+            androidProduct,
             'android',
+            'store',
+            downloadURL,
             androidRelease
         );
-    } else if (appStoreURL.test(downloadURL)) {
+    } else if (appStoreURL.test(downloadURL) || iTunesURL.test(downloadURL)) {
+        let iosProduct = 'unrecognized';
+        if (downloadURL.includes('/id989804926')) {
+            iosProduct = 'firefox';
+        } else if (downloadURL.includes('/id1055677337')) {
+            iosProduct = 'focus';
+        } else if (downloadURL.includes('/id1073435754')) {
+            iosProduct = 'klar';
+        } else if (downloadURL.includes('/id309601447')) {
+            iosProduct = 'pocket';
```

Comment:
Some of the logic here feels like it's getting quite complex. I know we chatted about using data attributes vs URL parsing. I'm happy to go with what you think works best here, but the additions to this file do give me pause for thought as to what's the most maintainable long term?

String.includes does not seem to be supported by IE unfortunately as well: https://caniuse.com/es6-string-includes

Reply:
It's only going to get more complex with the VPN support. 

The architecture of these events is just a long series of trade offs I'm not committed to this approach and data science has not given me any feedback on it one way or the other. Do you want to chat about it again?

Reply:
One option might be to trade adjust/app store referrals differently from Firefox Desktop and VPN file downloads. Though, that might just mean 2 complex files instead of 1.

Reply:
I'm not sure it's worth a big refactor as what you have works, and we can add tests to make sure things don't break. 

I guess my main worry is when we have to update a bouncer URL or app store link, and then forget to account for those changes here, then there's potential we'll create a bug. HTML data attributes might alleviate some of that risk, and make the logic here a bit less long winded. But that's not to say we couldn't also create bugs by not using attributes correctly (or consistently) either.

Reply:
I added query params to the adjust links like you mentioned on Slack. Let me know how it looks. I'm wondering if we should do it for the app/play store links too.